### ISSUE TITLE
TimberImage not hooking into all the right functions

### ISSUE THREAD

**Issue by sftsk** [timber/timber#1131]
### Expected behavior

We are using OffloadS3 on our Amazon server to keep images on S3. When accessing their url like this `TimberImage(icon).src` the path returned is not the right path. It gives the local path and not the s3 path (and the local file does not exist anymore as it is removed upon upload to s3).
`http://new.bpgen.com/app/uploads/2016/07/plate-with-53.jpg`

If I use a image size like here `TimberImage(icon).src('box-link-image')` it gives the right path: `http://s3-eu-west-1.amazonaws.com/bpg2-test/app/uploads/2016/07/20123314/plate-with-53-315x300.jpg`.

The problem is that this does not work with `svg` images as they won't be resized.

You can see this here: http://new.bpgen.com/how-to-order/

Could you please have a look at this and tell me if there is any workaround? It seems like you are not hooking into all the right function for getting the file url at all times. Also using `|resize` does not work for pretty much the same reason.

Thank you for your help,
Lukas

**Comment by sftsk** [comment]
I just saw that this has been discussed on #429 and that there have been some solutions.

Please look into this and fix this within Timber.
For the moment I have "fixed" it by applying the following filter:

```
// fix .src to use WordPress functions for attachment url (trigger s3 plugin hook)
function fix_timber_s3_src( $src, $ID ) {
	$size = '';
 	return wp_get_attachment_image_src($ID, $size)[0];
}
add_filter( 'timber_image_src', 'fix_timber_s3_src', 10, 2 );
```

would be great if you could pass size into the filter than this would almost be a working solution but in any case please fix this within TimberImage.

**Comment by jarednova** [comment]
Thanks @sftsk! I'll take a look at this, but feel free to submit a PR that takes care of the fix you're looking for

**Comment by jarednova** [comment]
@sftsk I believe this is now solved by #1320 — can you test against master to confirm? thanks!

**Comment by sftsk** [comment]
Hey @jarednova , I am sorry but I do not have control over that project anymore so I can not test it. Looking at the pull request it seems like everything is fine :)

Thanks a lot & great work!
